### PR TITLE
*: support `curdate()` as column's default value  | tidb-test=pr/2057

### LINKED ISSUE #38356: Date type Default Value Can't be (curdate())
## Bug Report
Date type Default Value Can't be (curdate()), But MySQL is OK.

### 1. Minimal reproduce step (Required)
CREATE TABLE DATE_TEST(
  id int,
  create_date date default (curdate())
)

### 2. What did you expect to see? (Required)
Query OK

### 3. What did you see instead (Required)
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 3 column 36 near "curdate()))"

### 4. What is your TiDB version? (Required)
6.0/6.1

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #38356

Problem Summary:

We don't support `curdate()` or `current_date()` as column's default value when we create/alter a table.

### What is changed and how it works?

Support it like `current_timestamp()`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/show.go [hunk 1]
```diff
@@ -1015,6 +1015,11 @@ func ConstructResultOfShowCreateTable(ctx sessionctx.Context, tableInfo *model.T
 					if col.GetDecimal() > 0 {
 						buf.WriteString(fmt.Sprintf("(%d)", col.GetDecimal()))
 					}
+				case "CURRENT_DATE":
+					buf.WriteString(" DEFAULT CURRENT_DATE")
+					if col.GetDecimal() > 0 {
+						buf.WriteString(fmt.Sprintf("(%d)", col.GetDecimal()))
+					}
```
Comment (xhebox) [inline executor/show.go:1022]: Maybe:

```
case "CURRENT_TIMESTAMP", "CURRENT_DATE":
  					buf.WriteString(" DEFAULT ")
                    buf.WriteString(defaultValue)
					if col.GetDecimal() > 0 {
						buf.WriteString(fmt.Sprintf("(%d)", col.GetDecimal()))
					}
```

File: expression/helper.go [hunk 2]
```diff
@@ -91,7 +91,7 @@ func GetTimeValue(ctx sessionctx.Context, v interface{}, tp byte, fsp int) (d ty
 	switch x := v.(type) {
 	case string:
 		upperX := strings.ToUpper(x)
-		if upperX == strings.ToUpper(ast.CurrentTimestamp) {
+		if upperX == strings.ToUpper(ast.CurrentTimestamp) || upperX == strings.ToUpper(ast.CurrentDate) {
```
Comment (xhebox) [inline expression/helper.go:94]: Or maybe we can also get the `lowerX` for comparison instead?

**Review by xhebox — APPROVED** [review]

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**